packages/vendure-plugin-primary-collection/test/e2e.spec.ts

```diff
@@ -0,0 +1,95 @@
+import { DefaultLogger, LogLevel, mergeConfig } from '@vendure/core';
+import {
+  createTestEnvironment,
+  registerInitializer,
+  SimpleGraphQLClient,
+  SqljsInitializer,
+  testConfig,
+} from '@vendure/testing';
+import { TestServer } from '@vendure/testing/lib/test-server';
+import { initialTestData } from './initial-test-data';
+import { testPaymentMethod } from '../../test/src/test-payment-method';
+import { PrimaryCollectionPlugin } from '../src/primary-collection-plugin';
+import { expect, describe, beforeAll, afterAll, it } from 'vitest';
+import { gql } from 'graphql-tag';
+
+describe('Product Primary Collection', function () {
+  let server: TestServer;
+  let adminClient: SimpleGraphQLClient;
+  let shopClient: SimpleGraphQLClient;
+  let serverStarted = false;
+
+  beforeAll(async () => {
+    registerInitializer('sqljs', new SqljsInitializer('__data__'));
+    const config = mergeConfig(testConfig, {
+      apiOptions: {
+        port: 3106,
+      },
+      logger: new DefaultLogger({ level: LogLevel.Debug }),
+      plugins: [PrimaryCollectionPlugin],
+      paymentOptions: {
+        paymentMethodHandlers: [testPaymentMethod],
+      },
+    });
+
+    ({ server, adminClient, shopClient } = createTestEnvironment(config));
+    await server.init({
+      initialData: {
+        ...initialTestData,
+        paymentMethods: [
+          {
+            name: testPaymentMethod.code,
+            handler: { code: testPaymentMethod.code, arguments: [] },
+          },
+        ],
+      },
+      productsCsvPath: './test/products.csv',
+      customerCount: 2,
+    });
+    serverStarted = true;
+  }, 60000);
+
+  it('Should start successfully', async () => {
+    expect(serverStarted).toBe(true);
+  });
+
+  const primaryCollectionQuery = gql`
+    query PrimaryCollectionQuery($productId: ID) {
+      product(id: $productId) {
+        name
+        primaryCollection {
+          id
+          name
+        }
+      }
+    }
+  `;
+
+  it("Should return 'Computers' as a primary collection for 'Laptop'", async () => {
+    const { product } = await shopClient.query(primaryCollectionQuery, {
+      productId: 'T_1',
+    });
+    expect(product.name).toBe('Laptop');
+    expect(product.primaryCollection.name).toBe('Computers');
+  });
+
+  it("Should return 'Computers' as a primary collection for 'Cars'", async () => {
```

Comment:
Can we make the test data a little more intuitive for this one?  'Computer' as primary collection for a 'Car' sounds like an unexpected answer :smile:. (Although these days most cars are sort of computers but...)